dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/citation/CitationDataExtractor.java

```diff
@@ -72,22 +75,56 @@ public CitationData create(FileMetadata fileMetadata, boolean direct) {
 
     private void extractAndWriteCommonValues(DatasetVersion dsv, CitationData data) {
         data.getAuthors().addAll(extractAuthors(dsv));
-        data.getProducers().addAll(dsv.getDatasetProducerNames());
+        data.getProducers().addAll(extractProducers(dsv));
+        data.getDistributors().addAll(dsv.getDatasetFieldValuesByTypeName(DatasetFieldConstant.distributorName));
         data.getFunders().addAll(dsv.getUniqueGrantAgencyValues());
         data.getKindsOfData().addAll(dsv.getKindOfData());
         data.getDatesOfCollection().addAll(dsv.getDatesOfCollection());
         data.getLanguages().addAll(dsv.getLanguages());
         data.getSpatialCoverages().addAll(dsv.getSpatialCoverages());
         data.getKeywords().addAll(dsv.getKeywords());
+        data.getOtherIds().addAll(dsv.getDatasetFieldValuesByTypeName(DatasetFieldConstant.otherIdValue));
 
         data.setDate(extractCitationDate(dsv))
                 .setYear(new SimpleDateFormat("yyyy").format(data.getDate()))
+                .setProductionPlace(extractField(dsv, DatasetFieldConstant.productionPlace))
+                .setProductionDate(extractProductionDate(dsv))
+                .setReleaseYear(extractReleaseYear(dsv))
+                .setRootDataverseName(dsv.getRootDataverseNameforCitation())
                 .setTitle(dsv.getTitle())
                 .setSeriesTitle(dsv.getSeriesTitle())
                 .setPublisher(extractPublisher(dsv))
                 .setVersion(extractVersion(dsv));
     }
 
+    private String extractField(DatasetVersion dsv, String typeName) {
+        return dsv.getDatasetFieldValueByTypeName(typeName)
+                .orElse(null);
+    }
+
+    private String extractProductionDate(DatasetVersion dsv) {
+        String rawDate = dsv.getDatasetFieldValueByTypeName(DatasetFieldConstant.productionDate)
+                .orElse(StringUtils.EMPTY);
+        Pattern year = Pattern.compile("\\d{4}");
+        Matcher yearMatcher = year.matcher(rawDate);
+        if (yearMatcher.find()) {
```

Comment:
It's because in this case we get the date as `String`, and in the other cases as `Date`. So it seems that the way with regex is cheaper than converting `String` to `Date` and the formatting the result.